### ISSUE TITLE
Error about the mask in ScaledDotProductAttention

### ISSUE THREAD

**Issue by yangze0930** [jadore801120/attention-is-all-you-need-pytorch#57]
Currently, the attention mask in the ScaledDotProductAttention is generated in Line 28 in Models.py by:
`pad_attn_mask = seq_k.data.eq(Constants.PAD).unsqueeze(1)`
`pad_attn_mask = pad_attn_mask.expand(mb_size, len_q, len_k)`

Ignoring the batch dimension for an explanation, I assume the generated pad_attn_mask is a matrix of shape (len_q * len_k), then this code will produce the matrix like **[A 1]**, where **1** is an all one submatrix. However, I think the generated attention mask should be like **[B 1 // 1 1]**, where **1** is an all one submatrix and **//** means line break (sorry I don't know how to type formula in Markdown environments).

**Comment by cherryyue** [comment]
Also have this problem.

**Comment by saitarslanboun** [comment]
So, why this error might lead to? I am currently using this repository code for my research. Do I have to retrain my models after fixing this?

**Comment by jadore801120** [comment]
Hello @yangze0930 @cherryyue , sorry I dont quite get your thought.
Would you mind to clarify it for me more detailedly? 
Thanks!
Yu-Hsiang

**Comment by seasa2016** [comment]
I think it might not be the problem since the embedding of the padding word is zero for all dimension.
Thus it is not important whether it is mask or not.

**Comment by yangze0930** [comment]
@jadore801120 @seasa2016 Thanks for your clarification, now I understand it.
